File: core-examples/src/main/rb/io/vertx/example/core/verticle/worker/workerverticle.rb
puts "[Worker] Starting in #{Java::JavaLang::Thread.current_thread().get_name()}"

$vertx.event_bus().consumer("sample.data") { |message|
  puts "[Worker] Consuming data in #{Java::JavaLang::Thread.current_thread().get_name()}"
  body = message.body().to_string()
  message.reply(body.to_upper_case())
}

File: core-examples/src/main/js/io/vertx/example/core/verticle/worker/worker_verticle.js
console.log("[Worker] Starting in " + Java.type("java.lang.Thread").currentThread().getName());

vertx.eventBus().consumer("sample.data", function (message) {
  console.log("[Worker] Consuming data in " + Java.type("java.lang.Thread").currentThread().getName());
  var body = message.body().toString();
  message.reply(body.toUpperCase());
});
